tests/manual/events/KeyboardTest.java

```diff
@@ -0,0 +1,812 @@
+
+            builder.addAbsent(KeyCode.PLUS);
+
+            return builder.getList();
+        }
+
+        /* The German QWERTZ layout */
+        public static KeyList germanKeys() {
+            KeyListBuilder builder = new KeyListBuilder();
+            builder.addCommon();
+            builder.addLetters();
+
+            /* Include one combination that involves Option/AltGr */
+            final String altGrSeven = (onMac ? "|" : "{");
+            final String decimalCharacter = (onLinux ? "." : ",");
+
+            builder.add(KeyCode.DIGIT0, "0", "=");
+            builder.add(KeyCode.DIGIT1, "1", "!");
+            builder.add(KeyCode.DIGIT2, "2", DOUBLE_QUOTE);
+            builder.add(KeyCode.DIGIT3, "3", SECTION);
+            builder.add(KeyCode.DIGIT4, "4", "$");
+            builder.add(KeyCode.DIGIT5, "5", "%");
+            builder.add(KeyCode.DIGIT6, "6", "&");
+            builder.add(KeyCode.DIGIT7, "7", "/", altGrSeven);
+            builder.add(KeyCode.DIGIT8, "8", "(");
+            builder.add(KeyCode.DIGIT9, "9", ")");
+
+            builder.add(KeyCode.LESS,        "<", ">");
+            builder.add(KeyCode.PLUS,        "+", "*");
+            builder.add(KeyCode.NUMBER_SIGN, "#", QUOTE);
+            builder.add(KeyCode.COMMA,       ",", ";");
+            builder.add(KeyCode.PERIOD,      ".", ":");
+            builder.add(KeyCode.MINUS,       "-", "_");
+
+            builder.add(KeyCode.DECIMAL,     decimalCharacter);
+
+            builder.addAbsent(KeyCode.COLON);
+
+            return builder.getList();
+        }
+
+        /*
+         * A generic Latin layout. No digits since layouts derived from French
+         * won't generate digit characters and may not even be encoded as digits
+         * on Linux.
+         */
+        public static KeyList latinKeys() {
+            KeyListBuilder builder = new KeyListBuilder();
+            builder.addCommon();
+            builder.addLetters();
+            return builder.getList();
+        }
+
+        /*
+         * For non-Latin layouts (Greek, Cyrillic) we should be able to access
+         * the letter KeyCodes though we have no idea what characters they
+         * generate.
+         */
+        public static KeyList nonLatinKeys() {
+            KeyListBuilder builder = new KeyListBuilder();
+            builder.addCommon();
+            builder.addDigits();
+            for (Character c = 'A'; c <= 'Z'; ++c) {
+                String s = String.valueOf(c);
+                KeyCode code = KeyCode.valueOf(s);
+                builder.add(code, "wild");
+            }
+            return builder.getList();
+        }
+    }
+
+    private enum Layout {
+        US_ENGLISH("U.S. English", KeyListBuilder.usEnglishKeys()),
+        FRENCH("French", KeyListBuilder.frenchKeys()),
+        GERMAN("German", KeyListBuilder.germanKeys()),
+        LATIN("Latin", KeyListBuilder.latinKeys()),
+        NON_LATIN("non-Latin", KeyListBuilder.nonLatinKeys());
```

Comment:
How hard would be adding another languages to this test, like Spanish? I see that you already commented that the robot can't access dead keys (so no possible test for `ñ` for instance...)?

Reply:
Thanks for adding Spanish, I've tested it successfully.